pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotBrokerDebug.java

```diff
@@ -114,28 +116,55 @@ public TimeBoundaryInfo getTimeBoundary(
   public Map<String, Map<ServerInstance, List<String>>> getRoutingTable(
       @ApiParam(value = "Name of the table") @PathParam("tableName") String tableName) {
     Map<String, Map<ServerInstance, List<String>>> result = new TreeMap<>();
+    getRoutingTable(tableName, (tableNameWithType, routingTable) -> result.put(tableNameWithType,
+        routingTable.getServerInstanceToSegmentsMap(false)));
+    if (!result.isEmpty()) {
+      return result;
+    } else {
+      throw new WebApplicationException("Cannot find routing for table: " + tableName, Response.Status.NOT_FOUND);
+    }
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/v2/debug/routingTable/{tableName}")
```

Comment:
Not a big fan of `v2`
```suggestion
  @Path("/debug/routingTableWithOptionalSegments/{tableName}")
```